hi-ml-histopathology/src/histopathology/configs/classification/BaseMIL.py

```diff
@@ -59,47 +62,36 @@ class BaseMIL(LightningContainer):
     batch_size: int = param.Integer(16, bounds=(1, None), doc="Number of slides to load per batch.")
-    max_bag_size: int = param.Integer(1000, bounds=(0, None),
-                                      doc="Upper bound on number of tiles in each loaded bag during training stage. "
-                                          "If 0 (default), will return all samples in each bag. "
-                                          "If > 0, bags larger than `max_bag_size` will yield "
-                                          "random subsets of instances.")
-    max_bag_size_inf: int = param.Integer(0, bounds=(0, None),
-                                          doc="Upper bound on number of tiles in each loaded bag during "
-                                          "validation and test stages."
-                                          "If 0 (default), will return all samples in each bag. "
-                                          "If > 0 , bags larger than `max_bag_size_inf` will yield "
-                                          "random subsets of instances.")
-    cache_mode: CacheMode = param.ClassSelector(default=CacheMode.MEMORY, class_=CacheMode,
-                                                doc="The type of caching to perform: "
-                                                    "'memory' (default), 'disk', or 'none'.")
-    precache_location: CacheLocation = param.ClassSelector(default=CacheLocation.CPU, class_=CacheLocation,
-                                                           doc="Whether to pre-cache the entire transformed dataset "
-                                                               "upfront and save it to disk and if re-load in cpu or "
-                                                               "gpu. Options: `none`,`cpu` (default), `gpu`")
     encoding_chunk_size: int = param.Integer(0, doc="If > 0 performs encoding in chunks, by loading"
                                                     "enconding_chunk_size tiles per chunk")
-    is_caching: bool = param.Boolean(False, doc="If True, cache the encoded tile features "
-                                     "(disables random subsampling of tiles). "
-                                     "If False (default), load the tiles without caching "
-                                     "(enables random subsampling of tiles).")
     # local_dataset (used as data module root_path) is declared in DatasetParams superclass
 
+    def __init__(self, **kwargs: Any) -> None:
+        super().__init__(**kwargs)
+        self.best_checkpoint_filename = "checkpoint_max_val_auroc"
+        self.best_checkpoint_filename_with_suffix = self.best_checkpoint_filename + ".ckpt"
+        self.checkpoint_folder_path = f"{self.outputs_folder}/checkpoints/"
+        self.callbacks = self.get_callbacks()
+
     @property
     def cache_dir(self) -> Path:
-        raise NotImplementedError
+        return Path(f"himl_cache/{self.__class__.__name__}-{self.encoder_type}/")
```

Comment:
Do the cache results actually end up in the AML outputs? If so, we should find a better place for the cached data